### PR TITLE
Fix underlying type of enums

### PR DESCRIPTION
**Pull request by kornilova-l**
Issue #12

### CODE REVIEW

File: ir/Enum.cpp [hunk 1]
```diff
@@ -13,27 +12,51 @@ uint64_t Enumerator::getValue() {
...
+        if (enumerator.getValue() > std::numeric_limits<int>::max()) {
+            s << "L";
+        }
+        if (e.type == "native.CUnsignedInt") {
+            s << ".toUInt" << "\n";
+        } else if (e.type == "native.CUnsignedLong") {
+            s << ".toULong" << "\n";
+        } else {
+            llvm::errs() << "Enum type is unsupported: " << e.type << "\n";
+            llvm::errs().flush();
```
Comment (kornilova-l) [inline ir/Enum.cpp]: This should not happen since `uint64_t` is `unsigned long int`

**Comment by kornilova-l** [comment]
@jonas, I added types to enumerators, they used to be regular Scala types, now they are `native.CUnsignedInt` or `native.CUnsignedLong`, see [tests/samples/Enum.scala](https://github.com/kornilova-l/scala-native-bindgen/blob/de446f062775f92e1592a7fdb626ff70d01bc468/tests/samples/Enum.scala)

File: tests/samples/Typedef.scala [hunk 2]
```diff
@@ -9,21 +9,21 @@ object Typedef {
...
-  final val enum_days_FRIDAY = 4
-  final val enum_days_SATURDAY = 5
-  final val enum_days_SUNDAY = 6
+  final val enum_days_MONDAY: enum_days = 0.toUInt
+  final val enum_days_TUESDAY: enum_days = 1.toUInt
+  final val enum_days_WEDNESDAY: enum_days = 2.toUInt
+  final val enum_days_THURSDAY: enum_days = 3.toUInt
+  final val enum_days_FRIDAY: enum_days = 4.toUInt
+  final val enum_days_SATURDAY: enum_days = 5.toUInt
+  final val enum_days_SUNDAY: enum_days = 6.toUInt
```
Comment (jonas) [inline tests/samples/Typedef.scala]: 👍

File: tests/samples/Enum.h [hunk 3]
```diff
@@ -7,3 +7,12 @@ enum days {
     SATURDAY = 3,
     SUNDAY // = 4
 };
+
+enum bigValues {
+    A = 10000000000, // does not fit into int
+    B = 1
```
[the remark this answers is not in the document]
Comment (kornilova-l) [reply tests/samples/Enum.h]: This is a good point, I totally forgot about negative numbers

**Pull request merged**